doctr/models/detection/linknet/base.py

```diff
@@ -156,78 +156,93 @@ class _LinkNet(BaseModel):
 
     def build_target(
         self,
-        target: List[np.ndarray],
-        output_shape: Tuple[int, int],
+        target: List[Dict[str, np.ndarray]],
+        output_shape: Tuple[int, int, int],
     ) -> Tuple[np.ndarray, np.ndarray]:
+        """Build the target, and it's mask to be used from loss computation.
 
-        if any(t.dtype != np.float32 for t in target):
+        Args:
+            target: target coming from dataset
+            output_shape: shape of the output of the model without batch_size
+
+        Returns:
+            the new formatted target and the mask
+        """
+
+        if any(t.dtype != np.float32 for tgt in target for t in tgt.values()):
             raise AssertionError("the expected dtype of target 'boxes' entry is 'np.float32'.")
-        if any(np.any((t[:, :4] > 1) | (t[:, :4] < 0)) for t in target):
+        if any(np.any((t[:, :4] > 1) | (t[:, :4] < 0)) for tgt in target for t in tgt.values()):
             raise ValueError("the 'boxes' entry of the target is expected to take values between 0 & 1.")
 
-        h, w = output_shape
-        target_shape = (len(target), h, w, 1)
+        h: int
+        w: int
+        if is_tf_available():
```

Comment:
We should avoid the permanent usage of torch / tf available better:

check shape: channels last => Tf / channels first => PT   (especially if we think further -> Onnx this conditions have to be removed completly if we want to reuse Pre-/Postprocessing

Reply:
Added the info from where the method was used.